pyspark-iris/{{ cookiecutter.repo_name }}/src/{{ cookiecutter.python_package }}/context.py

```diff
@@ -27,12 +27,23 @@
 # limitations under the License.
 
 """Entry point for running a Kedro pipeline as a Python package."""
+import logging
 from pathlib import Path
 from typing import Any, Dict, Union
 
 from kedro.framework.context import KedroContext
-from pyspark import SparkConf
-from pyspark.sql import SparkSession
+
+logger = logging.getLogger(__name__)
+
+try:
+    from pyspark import SparkConf
+    from pyspark.sql import SparkSession
+except ModuleNotFoundError:
+    logger.warning(
+        "Could not import pyspark. Please ensure pyspark is installed "
+        "before running any pipeline. You can use 'kedro install' to "
+        "install project dependencies."
```

Comment:
Note: this warning will show when `kedro install` is called for the first time as well, so the wording is rather important (reason why I added "_before running any pipeline_" to hopefully indicate that this is not `kedro install` failing).

As it is now, this is the full log a user would see upon calling `kedro install`:
```
(pyspark_test) ignacio@A-PC19VW0T:~/pyspark-test$ kedro install
2021-07-27 12:18:28,458 - pyspark_test.context - WARNING - Could not import pyspark. Please ensure pyspark is installed before running any pipeline. You can use 'kedro install' to install project dependencies.
2021-07-27 12:18:30,087 - root - INFO - Registered CLI hooks from 1 installed plugin(s): kedro-telemetry-0.1.1
Kedro-Telemetry is installed, but you have opted out of sharing usage analytics so none will be collected.
No requirements.in found. Copying contents from requirements.txt...
/home/ignacio/miniconda3/envs/pyspark_test/bin/python3.7 -m piptools compile -q /home/ignacio/pyspark-test/src/requirements.in
/home/ignacio/miniconda3/envs/pyspark_test/bin/python3.7 -m pip install -U -r /home/ignacio/pyspark-test/src/requirements.txt
[...]
Installing collected packages: [...]
[...]
Successfully installed [...]
Requirements installed!
```